api/pkg/apis/v1alpha1/utils/utils.go

```diff
@@ -321,6 +322,35 @@ func jsonPathQuery(obj interface{}, jsonPath string) (interface{}, error) {
 	}
 }
 
+func JsonParseProperty(properties map[string]interface{}, fieldPath string) (any, bool) {
+	query, err := gojq.Parse(formatPathForNestedJsonField(fieldPath))
+	if err != nil {
+		return nil, false
+	}
+
+	var value any
+	iter := query.Run(properties)
+	for {
+		result, ok := iter.Next()
+		if !ok {
+			// iterator terminates
+			break
+		}
+		if _, ok := result.(error); ok {
+			return nil, false
+		}
+		value = result
+	}
+	return value, value != nil
+}
+
+func formatPathForNestedJsonField(s string) string {
+	if len(s) == 0 || (len(s) > 0 && s[0] == '.') {
```

Comment:
You can just write len(s)==0 || s[0] == '.'

Reply:
removed this line to adopt jq syntax.